apps/src/applab/designElements/textInput.jsx

```diff
@@ -151,5 +151,21 @@ module.exports = {
     element.style.backgroundColor = '';
 
     return element;
+  },
+
+  onDeserialize: function(element) {
+    $(element).on('mousedown', function(e) {
+      if (!Applab.isRunning()) {
+        // Disable clicking into text input unless running
+        e.preventDefault();
+      }
+    });
+
+    // swallow keydown unless we're running
+    $(element).on('keydown', function (e) {
+      if (!Applab.isRunning()) {
+        e.preventDefault();
+      }
+    });
   }
```

Comment:
it is sometimes possible to focus a text input or text area by clicking a nearby element and tabbing into it. Rather than removing the focus or making sure this never happens, we just ignore the keystrokes if it does somehow get the focus.